src/core/writer.js

```diff
@@ -32,7 +32,7 @@ async function writeObject(ref, obj, buffer, { encrypt = null }) {
     await writeDict(obj, buffer, transform);
   } else if (obj instanceof BaseStream) {
     await writeStream(obj, buffer, transform);
-  } else if (Array.isArray(obj)) {
+  } else if (Array.isArray(obj) || ArrayBuffer.isView(obj)) {
```

Comment:
Here, and below, can't we use the existing [`isArrayBuffer` helper function](https://github.com/mozilla/pdf.js/blob/1cdbcfef821c7f6e81ea22fe68a8b815bca01c4e/src/shared/util.js#L966-L968) instead?

Reply:
Why do we have this `isArrayBuffer` function ? Why not just using `ArrayBuffer.isView` ?